src/lib/com/yetanalytics/xapipe/filter.clj

```diff
@@ -195,59 +190,46 @@
   :ret pattern-filter-pred-spec)
 
 (defn pattern-filter-pred
-  "Given a list of profile URLs generate a stateful predicate to filter
-  to only statements in patterns. If pattern ids are provided, limit filtered
-  statements to those."
   [{:keys [profile-urls
            pattern-ids]}]
-  (let [fsm-map (-> profile-urls
-                    (->> (map get-profile)
-                         (map per/profile->fsms)
-                         (into {}))
-                    (cond->
-                        (not-empty pattern-ids)
-                      (select-keys pattern-ids)))]
+  (let [fsm-map (apply
+                 per/compile-profiles->fsms
+                 (map get-profile profile-urls)
+                 (cond-> []
+                   (not-empty pattern-ids)
+                   (conj :selected-patterns pattern-ids)))]
```

Comment:
Doesn't `apply` incur a slight performance penalty compared to passing the args directly?

Reply:
If it does, it's likely negligible and this is at init time where I don't care very much

Reply:
Oh yeah if it's at init time then we really do not care.

Reply:
It is true that in late versions of clojure I could just use a map for the kwargs, but honestly it doesn't matter, and `apply` is idiomatic + readable

Reply:
I was curious so I checked:
```clojure

(defn apply
  "Applies fn f to the argument list formed by prepending intervening arguments to args."
  {:added "1.0"
   :static true}
  ([^clojure.lang.IFn f args]
     (. f (applyTo (seq args))))
  ([^clojure.lang.IFn f x args]
     (. f (applyTo (list* x args))))
  ([^clojure.lang.IFn f x y args]
     (. f (applyTo (list* x y args))))
  ([^clojure.lang.IFn f x y z args]
     (. f (applyTo (list* x y z args))))
  ([^clojure.lang.IFn f a b c d & args]
     (. f (applyTo (cons a (cons b (cons c (cons d (spread args)))))))))

```
Looks like apply does a little optimizing, More in the applicative interface I'm sure